torch/testing/_internal/common_methods_invocations.py

```diff
@@ -78,17 +78,52 @@ def __init__(self, cls_name=None, test_name=None, *,
                          test_name=test_name, device_type=device_type, dtypes=dtypes,
                          active_if=active_if)
 
+
+class SampleInputCtx(object):
+    """Context manager wrapping a SampleInput.
+
+    The end user can get the underlying SampleInput by entering a with
+    statement, e.g.:
+
+        with ctx as sample:
+            print(sample.args)
+            print(sample.input)
+
+    If case of exception, the context manager will prepend the name of the
+    SampleInput to the error message, making it possible to know which is the
+    specific sample which caused the test to fail.
+
+    If really needed, it is possible to manually the get underlying
+    SampleInput by calling get_sample(), but it is strongly adivsed to use the
+    with statement when possible, to get a better error reporting.
+    """
+    def __init__(self, sample):
+        assert isinstance(sample, SampleInput)
+        self._sample = sample
+
+    def get_sample(self):
+        return self._sample
+
+    def __enter__(self):
+        return self._sample
+
+    def __exit__(self, etype, evalue, tb):
+        if etype:
+            evalue.args = (f'[SampleInput: {self._sample.name}] {evalue}',)
```

Comment:
What if the sample input doesn't have a name?

Reply:
it shouldn't happen:
1. `SampleInput.__init__` sets `self.name` to `None` if none is given, so the line above never crashes
2. but most importantly, `OpInfo.sample_inputs()` automatically assigns a `name` based on the index, in case the name was none. So well-behaved code which uses the official API will always get named `SampleInput`s.